crates/lib/src/address.rs

```diff
@@ -188,8 +217,39 @@ mod tests {
     use super::*;
     use proptest::prelude::*;
     use spacetimedb_sats::bsatn;
+    use spacetimedb_sats::ser::serde::SerializeWrapper;
     use spacetimedb_sats::GroundSpacetimeType as _;
 
+    #[test]
+    fn address_json_serialization_big_endian() {
+        let addr = Address::from_be_byte_array([0xff, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
+
+        let hex = addr.to_hex();
+        let json1 = serde_json::to_string(&addr).unwrap();
+        let json2 = serde_json::to_string(&AddressForUrl::from(addr)).unwrap();
+
+        assert!(
+            json1.contains(hex.as_str()),
+            "expected {json1} to contain {hex} but it didn't"
+        );
+        assert!(
+            json2.contains(hex.as_str()),
+            "expected {json2} to contain {hex} but it didn't"
+        );
+
+        // Serde made the slightly odd choice to serialize u128 as decimals in JSON.
+        // So we have an incompatibility between our formats here :/
+        // The implementation of serialization for `sats` types via `SerializeWrapper` just calls
+        // the `serde` implementation to serialize primitives, so we can't fix this
+        // unless we make a custom implementation of `Serialize` and `Deserialize` for `Address`.
```

Comment:
I would seriously consider doing this. Most JSON parsers do not correctly handle large integers, in particular `JSON.parse` in the browser does not do it correctly unless you use [special sauce](https://stackoverflow.com/questions/69644298/how-to-make-json-parse-to-treat-all-the-numbers-as-bigint). I'd be concerned about people trying to parse a message containing an Address and getting a float with lost information.

Reply:
Would we need a PR in all the SDKs or just in Typescript since it only affects JSON or in none of them because we use BSATN. Do you know? I tend to agree with your assessment. I just don't have a sense of what it might break.

Reply:
Yeah I'm not sure what the current status of the protocol formats is either. Can check.

Reply:
Everything is using BSATN at the moment as far as I can tell, including BSATN. So this wouldn't be a huge change. It would possibly affect `web`.

Reply:
...It's pretty tricky though, and it will be inconsistent with other `u128`s in the same JSON if that comes up.

I think it's fine to not do this for now. Isn't `Address` going away anyway?

Reply:
Address is getting replaced with `ConnectionId` which is also a `u128` so the problem won't be going away entirely.